docs/libcudacxx/extended_api/execution_model.rst

```diff
@@ -0,0 +1,261 @@
+.. _libcudacxx-extended-api-execution-model:
+
+Execution model
+===============
+
+CUDA C++ aims to provide `_parallel forward progress_ [intro.progress.9] <https://eel.is/c++draft/intro.progress#9>`__
+for all device threads of execution, facilitating the parallelization of pre-existing C++ applications with CUDA C++.
+
+.. dropdown:: [intro.progress]
+
+    - `[intro.progress.7] <https://eel.is/c++draft/intro.progress#7>`__: For a thread of execution
+      providing `concurrent forward progress guarantees <https://eel.is/c++draft/intro.progress#def:concurrent_forward_progress_guarantees>`__,
+      the implementation ensures that the thread will eventually make progress for as long as it has not terminated.
+
+      [Note 5: This applies regardless of whether or not other threads of execution (if any) have been or are making progress.
+      To eventually fulfill this requirement means that this will happen in an unspecified but finite amount of time. — end note]
+
+    - `[intro.progress.9] <https://eel.is/c++draft/intro.progress>`__: For a thread of execution providing
+      `parallel forward progress guarantees <https://eel.is/c++draft/intro.progress#9>`__, the implementation is not required to ensure that
+      the thread will eventually make progress if it has not yet executed any execution step; once this thread has executed a step,
+      it provides concurrent forward progress guarantees.
+
+      [Note 6: This does not specify a requirement for when to start this thread of execution, which will typically be specified by the entity
+      that creates this thread of execution. For example, a thread of execution that provides concurrent forward progress guarantees and executes
+      tasks from a set of tasks in an arbitrary order, one after the other, satisfies the requirements of parallel forward progress for these
+      tasks. — end note]
+
+.. _libcudacxx-extended-api-execution-model-host-threads:
+
```

Comment:
@ericniebler @mhoemmen does the following suggestion address this comment: https://github.com/NVIDIA/cccl/pull/3873/files#r1964097678

```suggestion

The CUDA C++ Programming Langue is an extension of the C++ Programming Language.
This section documents the modifications and extensions to the `[intro.progress] <https://eel.is/c++draft/intro.progress>`__ section of the current `ISO International Standard ISO/IEC 14882 – Programming Language C++ <https://eel.is/c++draft/>`__ draft. 
Modified sections are called out explicitly and their diff is shown in **bold**. 
All other sections are an addition.
```

Reply:
@gonzalobg Looks good!  This definitely addresses my concern.  A couple comments:

1. First line misspells "Language" once.
2. Given that the Working Draft is a continuously changing document, should we consider giving a date?  Alternately, should we use one of the `N????` drafts, that have a fixed text and date?

Reply:
We give it a date in the Memory Model docs (adjacent .rst to this one), and we haven't update those in a while. 

The current wording incorporates some C++26 changes like `while(1)` being well-defined. 

I think my preference would be to just say "draft" for now, and open an issue to fix this to C++26 when its released (and update the memory model docs to C++26 as well).

Reply:
I've created https://github.com/NVIDIA/cccl/issues/3898 to track this.